### PR TITLE
DM-42404: Run Felis validation on schemas

### PR DESCRIPTION
**Pull request by JeremyMcCormick**
This runs the new `felis validate` command on all SDM schemas in the `./yml` dir. The `felislint.yaml` workflow was removed and replaced by `validate.yaml`. Other workflows were also updated to be compatible with the current version of Felis. 

For now, Felis is installed via a github URL, as we are trying to reclaim the name from a dead package on PyPI. This can be changed to `pip install felis` when that is working properly.

### CODE REVIEW

**Review by cbanek — APPROVED** [review]
Don't you want to delete felislint.yaml?

Anyway, this all looks reasonable, but I do see there's a fail to run on the felislinter.yaml action?

**Comment by JeremyMcCormick** [comment]
> Don't you want to delete felislint.yaml?
> 
> 
> 
> Anyway, this all looks reasonable, but I do see there's a fail to run on the felislinter.yaml action?

**Comment by JeremyMcCormick** [comment]
> Don't you want to delete felislint.yaml?
> 
> Anyway, this all looks reasonable, but I do see there's a fail to run on the felislinter.yaml action?

I deleted this and made note of a few warnings which are not replicated by the Pydantic model.

https://jira.lsstcorp.org/browse/DM-42482

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install git+http://github.com/lsst/felis.git@main#egg=felis
```
[the remark this answers is not in the document]
Comment (JeremyMcCormick) [reply .github/workflows/validate.yaml:23]: > I thought we were talking about like a felis github action

That is a possibility, but it is only two steps to checkout felis and run the validation. I don't see any real need for an external workflow at the moment, but I will discuss this with @gpdf.

File: .github/workflows/yamllint.yaml [hunk 2]
```diff
@@ -2,9 +2,6 @@ name: Lint YAML Files
 
 on:
   push:
```
[the remark this answers is not in the document]
Comment (JeremyMcCormick) [reply .github/workflows/yamllint.yaml]: I've changed the syntax to `on: push` for all the relevant workflows, which is the recommended syntax in the Github doc.

**Comment by JeremyMcCormick** [comment]
> Don't you want to delete felislint.yaml?
> 
> Anyway, this all looks reasonable, but I do see there's a fail to run on the felislinter.yaml action?

I deleted this and made note of a few warnings which are not replicated by the Pydantic model.

https://jira.lsstcorp.org/browse/DM-42482

**Pull request merged**